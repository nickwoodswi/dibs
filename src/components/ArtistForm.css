.artist-form{
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    background-color: rgb(0, 0, 0, .9);
    max-width: 600px;
    margin-left: auto;
    margin-right: auto;
    padding: 15px;
    color: white;
    font-size: 24px;
}
    .hold-type-header{
        width: 100%;
        text-align: center;
        font-style: italic;
    }
    .hold-type-buttons{
        display: flex;
        flex-wrap: wrap;
        justify-content: center;
        width: 100%;
        margin-left: auto;
        margin-right: auto;
        margin: 15px;
    }
        .hold_type{
            display: flex;
            font-size: 16px;
            align-items: center;
            justify-content: center;
            width: 150px;
            height: 75px;
            margin: 5px;
            background-color: white;
            border: none;
            color: black;
            padding: 5px;
        }
            .hold_type:hover, .active div{
                background-color: rgb(255, 0, 0, .8);
                color: white;
            }
    .range-date-selector{
        width: 100%;
        text-align: center;
        border-radius: 5px;
        border: 1px dotted red;
        padding: 15px;
    }
.performance-type-detail{
    width: 100%;
    margin-top: 15px;
    margin-bottom: 15px;
    padding: 15px;
}
    .performer-type{
        display: flex;
        width: 100%;
        font-size: 16px;
        padding: 5px;
        text-align: left;
    }
    input[type="radio"]{
        margin-left: 0px;
    }
.media{
    width: 100%;
    padding: 15px;
}
.support-need{
    width: 100%;
    padding: 15px;
    border: 1px dotted red;
    border-radius: 5px;
    margin-top: 15px;
    margin-bottom: 15px;
}
    .support-radio{
        margin: 15px;
    }
.event-detail{
    padding: 15px;
}
#event-detail{
    height: 200px;
}
.place-hold-button-container{
    margin: 15px;
    width: 100%;
    display: flex;
    justify-content: center;
}
    .submit-button{
        background-color: white;
        padding: 15px;
        color: black;
    }
        .submit-button:hover{
            background-color: red;
            color: white;
        }
        .submit-button:active{
            background-color: black;
        }
textarea{
    width: 100%;
}