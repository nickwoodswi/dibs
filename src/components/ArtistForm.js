import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { Route } from 'react-router-dom'
import StartDateSelector from './StartDateSelector'
import EndDateSelector from './EndDateSelector'
import MultiHoldSelector from './MultiHoldSelector'
import SingleHold from './SingleHold'
import './ArtistForm.css'

class ArtistForm extends Component {
    static defaultProps = {
        appState: {}
    }
    render() {
        return(
            <div className="artist-form">
                <div className="hold-type-header">PLACING HOLD(S) ON:</div>
                <div className="hold-type-buttons">
                    <Link to="/artist/single"><button id="single" name="hold_type" value="single" onClick={e => this.props.handleState('hold_type', e.target.value)}>A SINGLE DATE</button></Link>
                    <Link to="/artist/range"><button id="range" name="hold_type" value="range" onClick={e => this.props.handleState('hold_type', e.target.value)}>A RANGE OF DATES</button></Link>
                    <Link to="/artist/multi"><button id="multiple" name="hold_type" value="multiple" onClick={e => this.props.handleState('hold_type', e.target.value)}>MULTIPLE DATES</button></Link>
                </div>
                <Route path="/artist/single">
                    <SingleHold 
                        appState={this.props.appState} 
                        handleSingleHold={this.props.handleSingleHold}
                        submitForm={this.props.submitForm} />
                </Route>
                <Route path="/artist/range">
                    <div className="range-date-selector">
                        <StartDateSelector 
                            appState={this.props.appState} 
                            handleState={this.props.handleState} 
                            submitFOrm={this.props.submitForm}/><br/>
                        <EndDateSelector 
                            appState={this.props.appState} 
                            handleState={this.props.handleState} />
                    </div>
                </Route>
                <Route path="/artist/multi">
                    <MultiHoldSelector 
                        appState={this.props.appState} 
                        handleState={this.props.handleState} 
                        addHold={this.props.addHold}
                        deleteHold={this.props.deleteHold}
                        submitForm={this.props.submitForm}/>
                </Route>
                <div className="performance-type-detail">
                FOR:
                    <input type="text" placeholder="Band, Artist, or Performer Title/Name" value={this.props.appState.contact_act} onChange={e => this.props.handleState('contact_act', e.target.value)} />
                    <input type="text" placeholder="Your Home ZIP Code" value={this.props.appState.org_zip} onChange={e => this.props.handleState('org_zip', e.target.value)} />
                    <input type="radio" id="band" name="performance_type" value="band" onChange={e => this.props.handleState('event_type', e.target.value)} />Band
                    <input type="radio" id="solo" name="performance_type" value="solo" onChange={e => this.props.handleState('event_type', e.target.value)} />Solo Performer
                    <input type="radio" id="dj" name="performance_type" value="dj" onChange={e => this.props.handleState('event_type', e.target.value)} />DJ
                </div>

                <div className="media">
                Please link to your music, or a look at your performance:
                    <input type="text" value={this.props.appState.spotify} placeholder="Spotify Artist ID" onChange={e => this.props.handleState('spotify', e.target.value)} />
                    <input type="text" value={this.props.appState.youtube} type="text" placeholder="YouTube Channel or Video" onChange={e => this.props.handleState('youtube', e.target.value)} />
                    <input type="text" value={this.props.appState.facebook} type="text" placeholder="Facebook Page" onChange={e => this.props.handleState('facebook', e.target.value)} />
                    <input type="text" value={this.props.appState.bandcamp} type="text" placeholder="Bandcamp" onChange={e => this.props.handleState('bandcamp', e.target.value)} />
                    <input type="text" value={this.props.appState.website} type="text" placeholder="Website" onChange={e => this.props.handleState('website', e.target.value)} />
                </div>

                <div className="support-need">
                Do you need us to book local support?
                    <input type="radio" id="support_need" name="support_need" value="yes" onChange={e => this.props.handleState('support_need', e.target.value)} />Yes
                    <input type="radio" id="support_need" name="support_need" value="no" onChange={e => this.props.handleState('support_need', e.target.value)} />No
                </div>
                
                <div className="event-detail">
                    Provide any specific technical needs or details here:
                    <textarea id="event-detail" value={this.props.appState.event_detail} type="text" onChange={e => this.props.handleState('event_detail', e.target.value)}></textarea>
                </div>

                <div className="place-hold-button-container">
                    <button onClick={e => this.props.submitForm()}>PLACE HOLDS</button>
                </div>

            </div>
        )
    }
}

export default ArtistForm