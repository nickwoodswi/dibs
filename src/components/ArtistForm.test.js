import React from 'react';
import { render } from '@testing-library/react';
import ArtistForm from './ArtistForm';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders artist form', () => {
    render(<MemoryRouter><ArtistForm /></MemoryRouter>);
});

it('ArtistForm renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><ArtistForm /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});