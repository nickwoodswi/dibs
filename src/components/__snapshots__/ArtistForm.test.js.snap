// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistForm renders the UI as expected 1`] = `
<div
  className="artist-form"
>
  <div
    className="hold-type-header"
  >
    PLACING HOLD(S) ON:
  </div>
  <div
    className="hold-type-buttons"
  >
    <a
      href="/artist/single"
      onClick={[Function]}
    >
      <button
        id="single"
        name="hold_type"
        onClick={[Function]}
        value="single"
      >
        A SINGLE DATE
      </button>
    </a>
    <a
      href="/artist/range"
      onClick={[Function]}
    >
      <button
        id="range"
        name="hold_type"
        onClick={[Function]}
        value="range"
      >
        A RANGE OF DATES
      </button>
    </a>
    <a
      href="/artist/multi"
      onClick={[Function]}
    >
      <button
        id="multiple"
        name="hold_type"
        onClick={[Function]}
        value="multiple"
      >
        MULTIPLE DATES
      </button>
    </a>
  </div>
  <div
    className="performance-type-detail"
  >
    FOR:
    <input
      onChange={[Function]}
      placeholder="Band, Artist, or Performer Title/Name"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Your Home ZIP Code"
      type="text"
    />
    <input
      id="band"
      name="performance_type"
      onChange={[Function]}
      type="radio"
      value="band"
    />
    Band
    <input
      id="solo"
      name="performance_type"
      onChange={[Function]}
      type="radio"
      value="solo"
    />
    Solo Performer
    <input
      id="dj"
      name="performance_type"
      onChange={[Function]}
      type="radio"
      value="dj"
    />
    DJ
  </div>
  <div
    className="media"
  >
    Please link to your music, or a look at your performance:
    <input
      onChange={[Function]}
      placeholder="Spotify Artist ID"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="YouTube Channel or Video"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Facebook Page"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Bandcamp"
      type="text"
    />
    <input
      onChange={[Function]}
      placeholder="Website"
      type="text"
    />
  </div>
  <div
    className="support-need"
  >
    Do you need us to book local support?
    <input
      id="support_need"
      name="support_need"
      onChange={[Function]}
      type="radio"
      value="yes"
    />
    Yes
    <input
      id="support_need"
      name="support_need"
      onChange={[Function]}
      type="radio"
      value="no"
    />
    No
  </div>
  <div
    className="event-detail"
  >
    Provide any specific technical needs or details here:
    <textarea
      id="event-detail"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="place-hold-button-container"
  >
    <button
      onClick={[Function]}
    >
      PLACE HOLDS
    </button>
  </div>
</div>
`;
